zome.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (C) 2013-2017, The MetaCurrency Project (Eric Harris-Braun, Arthur Brock, et. al.)
+// Use of this source code is governed by GPLv3 found in the LICENSE file
+//---------------------------------------------------------------------------------------
+
+// Data integrity engine for distributed applications -- a validating monotonic
+//
+// Zome struct encapsulates logically related code, from "chromosome"
+
+package holochain
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+type Zome struct {
+	Name        string
+	Description string
+	Code        string // file name of DNA code
+	CodeHash    Hash
+	Entries     map[string]EntryDef
+	NucleusType string
+	Functions   map[string]FunctionDef
+
+	// cache for code and nucleus
+	code string
+	nuc  *Nucleus
+}
+
+// EntryDef struct holds an entry definition
+type EntryDef struct {
+	Name       string
+	DataFormat string
+	Schema     string // file name of schema or language schema directive
+	SchemaHash Hash
+	Sharing    string
+	validator  SchemaValidator
+}
+
+// FunctionDef holds the name and calling type of an DNA exposed function
+type FunctionDef struct {
+	Name        string
+	CallingType string
+}
+
+// ZomePath returns the path to the zome dna data
+// @todo sanitize the name value
+func (h *Holochain) ZomePath(z *Zome) string {
+	return h.DNAPath() + "/" + z.Name
```

Comment:
This is where it decodes the zome from the dna, right? In other words, at some point in early steps, the zome got loaded into memory as go objects and serialized into the entry and written, but after that it is accessed here or not?

I can see it must also decode the objects by Unmarshalling from the entry, and it more typically would write this directory from this action. Maybe always.

If this is more or less right, can you tell me which hc commands do either/both of these actions?